File: src/main/java/lambda/evaluator/StackGenerator.java
package lambda.evaluator;

import lambda.core.datamodels.Stack;

interface StackGenerator {

    <T> Stack<T> generateNewStack(Class<T> clazz);

    static StackGenerator newInstance() {
        return new StackGenerator() {
            @Override
            public <T> Stack<T> generateNewStack(Class<T> clazz) {
                return new Stack<>();
            }
        };
    }
}

File: src/main/java/lambda/evaluator/StackGenerator.kt
package lambda.evaluator

import lambda.core.datamodels.Stack

internal interface StackGenerator {
    fun <T> generateNewStack(clazz: Class<T>): Stack<T>
}

internal class StackGeneratorImpl: StackGenerator {

    override fun <T> generateNewStack(clazz: Class<T>): Stack<T> {
        return Stack()
    }
}